contrib/roles/windows/validation/tasks/validate_service.yml

```diff
@@ -0,0 +1,20 @@
+---
+- name: Windows validate service | Get service info
+  win_service:
+    name: "{{ service_name }}"
+  register: service_stat
+
+- name: Windows validate service | Fail if the service doesn't exist
+  fail:
+    msg: "The service {{ service_name }} doesn't exist"
+  when: not service_stat.exists
+
+- name: Windows validate service | Fail if the service is not running
+  fail:
+    msg: "The service {{ service_name }} is not running. Current state: {{ service_stat.state }}"
+  when: service_stat.state != "running"
```

Comment:
Would this catch the case when the service is stuck in a restart loop due to a constant crash at startup?

Reply:
Sadly no, but I believe that 95% of the cases will be just fine with this check. For the other 5%, I believe that we need some extra PowerShell logic.

So, my [big PR](https://github.com/openvswitch/ovn-kubernetes/pull/511) to improve the Windows playbooks changed the failure actions like [this](https://github.com/openvswitch/ovn-kubernetes/blob/master/contrib/roles/windows/kubernetes/tasks/run_minion_init.yml#L84), to all the Windows services.

Every Windows service will have these failure actions `actions=restart/0/restart/0/run/30000`. This means that if the service fails, it will try and restart it only **twice** with no waiting in between. All the subsequent failures **after** the first two will have a 30 seconds waiting time in between restarts. The service restart is done via a custom PowerShell command that does log rotation as well.

Coming back to the case when the service will continuously fail. After 2 restarts, you'll get service restarted **only** once at 30 seconds. This means that this check will probably be safe for most cases as described above, unless you're that unlucky and the task gets executed when the service is started after a 30 seconds interval.

If you want, I can add an extra PowerShell task and poll the service status for 30 seconds (check the `running` state once a second).

Reply:
Sounds fair enough, looks good like this in that case, anyway there's also the additional check after minion-init which tests the connectivity by pinging the gateway IP.